chia/pools/pool_config.py

```diff
@@ -63,18 +63,23 @@ def load_pool_config(root_path: Path) -> List[PoolWalletConfig]:
 def add_auth_key(root_path: Path, config_entry: PoolWalletConfig, auth_key: G1Element):
     config = load_config(root_path, "config.yaml")
     pool_list = config["pool"].get("pool_list", [])
+    updated = False
     if pool_list is not None:
         for pool_config_dict in pool_list:
             try:
                 if (
                     G1Element.from_bytes(hexstr_to_bytes(pool_config_dict["owner_public_key"]))
                     == config_entry.owner_public_key
                 ):
-                    pool_config_dict["authentication_public_key"] = bytes(auth_key).hex()
+                    auth_key_hex = bytes(auth_key).hex()
+                    if pool_config_dict.get("authentication_public_key", "") != auth_key_hex:
```

Comment:
`None`, the default if the key is missing, will be unequal to any string so that's probably good enough.

```suggestion
                    if pool_config_dict.get("authentication_public_key") != auth_key_hex:
```